python/libsettings.pyx

```diff
@@ -148,25 +148,38 @@ cdef class Settings:
     def destroy(self):
         settings_destroy(&self.ctx)
 
-    def write(self, section, name, value):
+    def write(self, section, name, value, encoding='ascii'):
```

Comment:
Not very important at this point, and thus this comment can be ignored, but I have a feeling that the encoding could be a server-side setting from application perspective and this library could be the middle-man storing it. The settings change the behavior of the device after all, so the device would know best how it wants to be controlled and what are valid values.

Following this logic, this library could have an additional function that tells the expected encoding, and read/read_all/write wouldn't have an encoding parameter but try to use the expected encoding for string values, throwing an exception if it cannot be done.

However, if there are string settings that accept arbitrary bytes, then this logic wouldn't work well. Then it would be better to take input as bytes/bytearrays. But I am not aware of string settings needing arbitrary bytes.

Reply:
There's some merit to this but currently I'd say not relevant since at least on Piksi side everything is (presumed) to be in `ascii` domain AFAIK.